patroni/ha.py

```diff
@@ -1357,6 +1386,14 @@ def _run_cycle(self):
                 self.state_handler.reset_cluster_info_state(None, self.patroni.nofailover)
                 raise
 
+            # If the cluster has been configured with a static primary,
+            # and we are not that primary, then do not proceed.
+            if self.is_static_primary_configured() and not self.is_static_primary():
+                self.shutdown()
+                return 'patroni cluster is configured with a static primary, \
+                        and this node is not the primary, shutting down and \
+                        refusing to start'
```

Comment:
`self.shutdown()` will not cause Patroni to shut down, but I believe it should be.
That behavior could be achieved the same way as here: https://github.com/zalando/patroni/blob/d8d634125c547122af22646bd9988686e75adc99/patroni/ha.py#L1448-L1451

Reply:
See my [comment here](https://github.com/zalando/patroni/pull/2318#discussion_r952957565). The intention is not to shutdown Patroni, only to ensure that Postgres itself is not running.